# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1288608883 14400
#      Mon Nov 01 06:54:43 2010 -0400
# Node ID 2114867f2a50ba199eb1e1e6779c7b9ebc539892
# Parent  298a75c128ad1f4acd22dc5afec5bf4422fafab3
axis.m: Exclude hggroup {x,y,z}data properties when determing tight axis limits.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-11-01  Ben Abbott <bpabbott@mac.com>
+
+	* plot/axis.m: Exclude hggroup {x,y,z}data properties when determing
+	tight axis limits.
+
 2010-10-31  Rik  <octave@nomad.inbox5.com>
 
 	* strings/base2dec.m, strings/bin2dec.m, strings/dec2base.m,
 	  strings/dec2bin.m, strings/dec2hex.m, strings/hex2dec.m:
 	  Improve docstrings, use more descriptive variable names,
 	  add more tests for functions used to convert between bases.
 	  
 2010-10-31  Konstantinos Poulios  <logari81@googlemail.com>
diff --git a/scripts/plot/axis.m b/scripts/plot/axis.m
--- a/scripts/plot/axis.m
+++ b/scripts/plot/axis.m
@@ -303,16 +303,20 @@ function curr_axis = __axis__ (ca, ax, v
 
 endfunction
 
 function lims = __get_tight_lims__ (ca, ax)
 
   ## Get the limits for axis ("tight").
   ## AX should be one of "x", "y", or "z".
   kids = findobj (ca, "-property", strcat (ax, "data"));
+  ## Since contours set the cdata for the patches to the hggroup zdata property, exclude
+  ## hgroups when determining the tight limits.
+  hg_kids = findobj (ca, "-property", strcat (ax, "data"), "type", "hggroup");
+  kids = setdiff (kids, hg_kids);
   if (isempty (kids))
     ## Return the current limits.
     lims = get (ca, strcat (ax, "lim"));
   else
     data = get (kids, strcat (ax, "data"));
     if (iscell (data))
       data = data (find (! cellfun (@isempty, data)));
       if (! isempty (data))
